src/backend/distributed/worker/worker_data_fetch_protocol.c

```diff
@@ -791,9 +795,9 @@ FetchForeignTable(const char *nodeName, uint32 nodePort, StringInfo tableName)
 
 	/* fetch foreign file to this node in an idempotent manner */
 	localFilePath = makeStringInfo();
-	appendStringInfo(localFilePath, FOREIGN_CACHED_FILE_PATH, tableName->data);
+	appendStringInfo(localFilePath, FOREIGN_CACHED_FILE_PATH, tableName);
```

Comment:
I take it back, finally (after many times reading the call tree) realized that the table name here has the shard identifier in it, so we're probably good from a collision standpoint.